XMMS WOTS uses value after moving from it

Found by `clang-tidy`

```
/home/jack/work/botan/src/lib/pubkey/xmss/xmss_wots.cpp:148:54: warning: Method called on moved-from object 'params' [clang-analyzer-cplusplus.Mov\
e]
   : XMSS_WOTS_Base(std::move(params), wots_keysig_t(params.len()))
                                                     ^~~~~~~~~~~~
```

cc @reneme

--- a/src/lib/pubkey/xmss/xmss_wots.cpp
+++ b/src/lib/pubkey/xmss/xmss_wots.cpp
@@ -145,8 +145,9 @@ XMSS_WOTS_PrivateKey::XMSS_WOTS_PrivateKey(XMSS_WOTS_Parameters params,
                                            std::span<const uint8_t> private_seed,
                                            const XMSS_Address& adrs,
                                            XMSS_Hash& hash)
-   : XMSS_WOTS_Base(std::move(params), wots_keysig_t(params.len()))
+   : XMSS_WOTS_Base(std::move(params))
    {
+   m_key_data.resize(m_params.len());
    const auto r = hash.prf(private_seed, adrs.bytes());
 
    for(size_t i = 0; i < m_params.len(); ++i)

--- a/src/lib/pubkey/xmss/xmss_wots.h
+++ b/src/lib/pubkey/xmss/xmss_wots.h
@@ -31,6 +31,9 @@ typedef std::vector<secure_vector<uint8_t>> wots_keysig_t;
 class XMSS_WOTS_Base
    {
    public:
+      XMSS_WOTS_Base(XMSS_WOTS_Parameters params)
+         : m_params(std::move(params)) {}
+
       XMSS_WOTS_Base(XMSS_WOTS_Parameters params, wots_keysig_t key_data)
          : m_params(std::move(params)), m_key_data(std::move(key_data)) {}
 

